config_files/training/config_lorem_ipsum_span_masking_T5.yaml

```diff
@@ -0,0 +1,272 @@
+            num_replicas: ${settings.cuda_env.world_size}
+            shuffle: false
+            dataset:
+              instance_key: val_dataset
+              pass_type: BY_REFERENCE
+    collate_fn:
+      instance_key: collate_fn
+      pass_type: BY_REFERENCE
+
+eval_dataloaders:
+  - instance_key: val_dataloader
+    pass_type: BY_REFERENCE
+
+checkpoint_saving:
+  component_key: checkpoint_saving
+  variant_key: default
+  config:
+    checkpoint_saving_strategy:
+      component_key: checkpoint_saving_strategy
+      variant_key: save_k_most_recent_checkpoints_strategy
+      config:
+        k: -1   # -1 to save all checkpoints
+    checkpoint_saving_execution:
+      component_key: checkpoint_saving_execution
+      variant_key: fsdp
+      config:
+        checkpoint_path: ${settings.paths.checkpointing_path}
+        global_rank: ${settings.cuda_env.global_rank}
+        experiment_id: ${settings.experiment_id}
+        get_num_tokens_from_num_steps_callable:
+          component_key: number_conversion
+          variant_key: num_tokens_from_num_steps_callable
+          config:
+            num_ranks: ${settings.cuda_env.world_size}
+            local_micro_batch_size: ${settings.training.local_train_micro_batch_size}
+            sequence_length: ${settings.training.sequence_length}        
+
+# resolving class types via different enums sucks...
+loss_fn:
+  component_key: loss
+  variant_key: clm_cross_entropy_loss
+  config:
+    target_key: target_ids
+    prediction_key: logits
+
+wrapped_model:
+  component_key: model
+  variant_key: fsdp_wrapped
+  config:
+    model:
+      instance_key: model_raw
+      pass_type: BY_REFERENCE
+    sync_module_states: true
+    mixed_precision_settings: BF_16
+    sharding_strategy: FULL_SHARD
+    block_names: [LongT5Block]
+
+model_raw:
+  component_key: model
+  variant_key: huggingface_pretrained_encoder_decoder_model
+  config:
+    #model_type: AutoModelForCausalLM
+    #model_name: TinyLlama/TinyLlama_v1.1 
+    model_type: LongT5ForConditionalGeneration
+    model_name: google/long-t5-tglobal-base
+    sample_key: ${settings.referencing_keys.sample_key}
+    prediction_key: logits
+    huggingface_prediction_subscription_key: logits
+
+#scheduler:
+#  component_key: scheduler
+#  variant_key: step_lr
+#  config:
+#    optimizer:
+#      instance_key: optimizer
+#      pass_type: BY_REFERENCE
+#    step_size: 1
+#    gamma: 0.1
+
+scheduler:
+  component_key: scheduler
+  variant_key: onecycle_lr
+  config:
+    optimizer:
+      instance_key: optimizer
+      pass_type: BY_REFERENCE
+    max_lr: 6e-4
+    div_factor: 10
+    final_div_factor: 1
+    total_steps: 256
+    pct_start: 0.01
+    anneal_strategy: cos
+
+optimizer:  
+  component_key: optimizer
+  variant_key: adam_w
+  config:
+    lr: 0.0001
+    betas: [0.9, 0.95]
+    eps: 1e-8
+    weight_decay: 1e-1
+    # need to fix in huggingface_model.py to same spec as in gpt2_model.py
+    weight_decay_groups_excluded: []
```

Comment:
Need to check if this has to be filled

Reply:
Fixed, logic in the model is already there, just changed the config